### DISCUSSION TITLE
How to exclude field in dict(Sqlalchemy object will be convert to dict with vars)

### DISCUSSION THREAD

**Discussion by wuranxu** [fastapi/fastapi#6365] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class User(Base):
    name = Column(String(32))
    password = Column(String(32))

async def query_user():
    a = User()
    return json_encoder(a, exclude=["password"])
```

### Description

Hi, I have check the source code about json_encoder, I think the code doesn't pass `exclude` info into `json_encoder`, but if i return a pydantic model, It will be work(I guess), It may be designed like that. Could you help me to encode my sqlalchemy object?Thanks a lot!
![image](https://user-images.githubusercontent.com/11266738/161996810-08a43ec8-6771-4eb9-bbf5-50cd37607d9d.png)

### Operating System

Windows

### FastAPI Version

0.75.0

### Python Version

3.9

**Comment by ddanier** [comment, marked as answer]
I would suggest using this code:

```python
class User(Base):
    name = Column(String(32))
    password = Column(String(32))


class UserDTO(pydantic.BaseModel):
    name: str

    class Config:
        orm_mode = True


@route.get(..., response_model=UserDTO)
async def query_user():
    a = User()
    return UserDTO.from_orm(a)
```


Benefits:
* As you defined the `UserDTO` you have full control over the fields
* Using `response_model` will create a good OpenAPI spec
* Separating database and DTO (data transfer object) models will create a good separation of concerns

Of course creating those DTO models can be a lot to write down. If you struggle with this you could create a helper function using `pydantic.create_model(...)` to dynamically create this DTO. And you may integrate include/exclude options here. ;-)

**Comment by wuranxu** [comment]
> I would suggest using this code:
> 
> ```python
> class User(Base):
>     name = Column(String(32))
>     password = Column(String(32))
> 
> 
> class UserDTO(pydantic.BaseModel):
>     name: str
> 
>     class Config:
>         orm_mode = True
> 
> 
> @route.get(..., response_model=UserDTO)
> async def query_user():
>     a = User()
>     return UserDTO.from_orm(a)
> ```
> 
> Benefits:
> 
> * As you defined the `UserDTO` you have full control over the fields
> * Using `response_model` will create a good OpenAPI spec
> * Separating database and DTO (data transfer object) models will create a good separation of concerns
> 
> Of course creating those DTO models can be a lot to write down. If you struggle with this you could create a helper function using `pydantic.create_model(...)` to dynamically create this DTO. And you may integrate include/exclude options here. ;-)

Thanks for your suggestion very much! I'm too lazy to define DTO for response model. It will display redundant fields in the database, Thanks again! 👍
